### PR TITLE
aruha-351 modify log format

### PR DESCRIPTION
**Pull request by rcillo**
Our previous log format used to look like this:

```
Sep 15 15:34:51 ip-10-246-4-222 docker/f27592b3a708[817]: [2016-09-15 15:34:51.528] MIVVr0eASeby3Pl5vrtFKvQ3 boot - 6 DEBUG [http-nio-8080-exec-52] --- org.springframework.security.web.access.intercept.FilterSecurityInterceptor: RunAsManager did not change Authentication object
```

It has some problems:

1. the date time misses timezone.
2. %x NDC (flow-id) is not properly delimited. It prints "null" sometimes.
3. what does "boot" mean?
4. %X{context} prints nothing.
5. PID is not used because there is always a single process inside docker image.
6. log level comes after a variable number of patterns (if MDC %X is or is not printed, for example)
7. finally the message. Can be multiline.

We changed it so that an example would look like:

```
Sep 15 15:34:51 ip-10-246-4-222 docker/f27592b3a708[817]: [2016-09-16 09:38:23.053Z] [DEBUG] [QTElULMvw7hBcKg88Tls9Bte] [http-nio-8080-exec-1] [org.springframework.security.web.access.intercept.FilterSecurityInterceptor] --- RunAsManager did not change Authentication object
```

This way it's easier to parse and extract information.

We are also going to change the grok pattern to extract fields.

```
%{SYSLOGBASE2:syslogbase} \[%{TIMESTAMP_ISO8601:timestamp}\] \[%{WORD:loglevel}\] \[%{WORD:flowId}\] \[%{NOTSPACE:thread}\] \[%{JAVACLASS:javaClass}\] --- %{GREEDYDATA:logMessage}
```

Which results in the following json:

```json
{
  "syslogbase": [
    [
      "Sep 15 15:34:51 ip-10-246-4-222 docker/f27592b3a708[817]:"
    ]
  ],
  "timestamp": [
    [
      "Sep 15 15:34:51"
    ]
  ],
  "MONTH": [
    [
      "Sep"
    ]
  ],
  "MONTHDAY": [
    [
      "15",
      null,
      "15"
    ]
  ],
  "TIME": [
    [
      "15:34:51"
    ]
  ],
  "HOUR": [
    [
      "15",
      null,
      null,
      "15",
      null
    ]
  ],
  "MINUTE": [
    [
      "34",
      null,
      null,
      "34",
      null
    ]
  ],
  "SECOND": [
    [
      "51",
      null,
      "51.053"
    ]
  ],
  "timestamp8601": [
    [
      null
    ]
  ],
  "YEAR": [
    [
      null,
      "2016"
    ]
  ],
  "MONTHNUM": [
    [
      null,
      "09"
    ]
  ],
  "ISO8601_TIMEZONE": [
    [
      null,
      "Z"
    ]
  ],
  "SYSLOGFACILITY": [
    [
      null
    ]
  ],
  "facility": [
    [
      null
    ]
  ],
  "priority": [
    [
      null
    ]
  ],
  "logsource": [
    [
      "ip-10-246-4-222"
    ]
  ],
  "IPORHOST": [
    [
      "ip-10-246-4-222"
    ]
  ],
  "HOSTNAME": [
    [
      "ip-10-246-4-222"
    ]
  ],
  "IP": [
    [
      null
    ]
  ],
  "IPV6": [
    [
      null
    ]
  ],
  "IPV4": [
    [
      null
    ]
  ],
  "SYSLOGPROG": [
    [
      "docker/f27592b3a708[817]"
    ]
  ],
  "program": [
    [
      "docker/f27592b3a708"
    ]
  ],
  "pid": [
    [
      "817"
    ]
  ],
  "appTimestamp": [
    [
      "2016-09-15 15:34:51.053Z"
    ]
  ],
  "loglevel": [
    [
      "DEBUG"
    ]
  ],
  "flowId": [
    [
      "QTElULMvw7hBcKg88Tls9Bte"
    ]
  ],
  "thread": [
    [
      "http-nio-8080-exec-1"
    ]
  ],
  "javaClass": [
    [
      "org.springframework.security.web.access.intercept.FilterSecurityInterceptor"
    ]
  ],
  "logMessage": [
    [
      "RunAsManager did not change Authentication object"
    ]
  ]
}
```

### CODE REVIEW

**Comment by antban** [comment]
I have only one question - what is ` --- `? And what is the reason for it

**Comment by rcillo** [comment]
This is a separator to make it easier for humans to understand when the message starts, e.g. the non structured part of the log line.

File: src/main/resources/log4j.properties [hunk 1]
```diff
@@ -1,8 +1,6 @@
 log4j.rootCategory=INFO, CONSOLE
 
-PID=????
-LOG_LEVEL_PATTERN=%5p
```
Comment (rcillo) [inline src/main/resources/log4j.properties]: I considered this formatting to be useless since we have delimiters now.

**Comment by antban** [comment]
:+1:

**Pull request merged**